quarkScriptAgent.py

```diff
@@ -0,0 +1,144 @@
+import os
+import re
+
+from termcolor import colored
+
+from langchain_openai import ChatOpenAI
+from langchain.agents import tool, AgentExecutor
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain.agents.output_parsers.openai_tools import OpenAIToolsAgentOutputParser
+from langchain.agents.format_scratchpad.openai_tools import (
+    format_to_openai_tool_messages,
+)
+
+from quark.script import Rule, _getQuark, QuarkResult
+
+if "OPENAI_API_KEY" not in os.environ:
+    api_key = input("OpenAI API Key: ")
+    os.environ["OPENAI_API_KEY"] = api_key
+
+
+@tool
+def defineRuleByJson():
+    """
+    This instance defines the detection rule
+    by loading the rule from a JSON.
+    """
```

Comment:
This tool only loads the rule from a given path. We could keep its name and its docstring simple. For example,

```python
@tool
def loadRule(rule_path: str):
    """
    Given a rule path,
    this instance loads a rule from the rule path.
    """
    ...
```